app/src/main/java/tech/relaycorp/courier/domain/ObserveStorageUsage.kt

```diff
@@ -0,0 +1,38 @@
+package tech.relaycorp.courier.domain
+
+import kotlinx.coroutines.flow.combine
+import tech.relaycorp.courier.data.database.StoredMessageDao
+import tech.relaycorp.courier.data.disk.DiskStats
+import tech.relaycorp.courier.data.model.StorageUsage
+import tech.relaycorp.courier.data.preference.StoragePreferences
+import javax.inject.Inject
+import kotlin.math.min
+
+class ObserveStorageUsage
+@Inject constructor(
+    private val storedMessageDao: StoredMessageDao,
+    private val storagePreferences: StoragePreferences,
+    private val diskStats: DiskStats
+) {
+
+    fun observe() =
+        combine(
+            storedMessageDao.observeFullSize(),
+            observeMaxStorageAvailable()
+        ) { usedStorage, maxStorageAvailable ->
+            StorageUsage(usedStorage, maxStorageAvailable)
+        }
+
+    private fun observeMaxStorageAvailable() =
+        combine(
+            storagePreferences.getMaxStoragePercentage(),
+            diskStats.observeAvailableStorage(),
+            storedMessageDao.observeFullSize()
+        ) { maxStoragePercentage, availableStorage, usedStorage ->
```

Comment:
I think it's important to qualify each method and variable with the unit they represent, especially here where we're dealing with percentages and bytes. The same applies to this method (`observeMaxStorageAvailable`).

Reply:
You are right, but I'll probably use types instead of naming for that.